dnadoh/recombine.py

```diff
@@ -0,0 +1,50 @@
+"""Recombination of the genomes of two people
+to produce a third person. Other features are
+averaged with some noise.
+"""
+
+from typing import List
+
+def recombine(
+    person1: Person,
+    person2: Person,
+    reference_genome: str,
+    person_gen: PersonGenerator,
+    pid: str,
+    recombination_prob: List[float],
+    recombination_loci: List[int]
+    ) -> Person:
+    """Recombine two genomes"""
+    # Error checking
+    assert len(recombination_loci) == len(recombination_prob)
+    assert len(person1.genome) == len(person2.genome)
+    assert all(recombination_loci >= 0 & recombination_loci < len(person1.genome))
+    assert all(recombination_prob <= 1 & recombination_prob >= 0)
+
+    # Initializing
+    genome1 = person1.genome
+    genome2 = person2.genome
+
+    if 0 not in recombination_loci:
+        recombination_loci = [0] + recombination_loci
+        recombination_prob = [0.5] + recombination_prob
+    if len(genome1) not in recombination_loci:
+        recombination_loci = recombination_loci + [len(genome1)]
+        recombination_prob = recombination_prob + [0.5]
+
+    ## Ordering
+    ordered_prob = [prob for _, prob in sorted(zip(recombination_loci, recombination_prob))]
```

Comment:
do you need to re-sort in the line below given that you're sorting here? (trying to follow the logic)

Reply:
Yes, I sort the probabilities according to the order of the loci first, and then I sort the loci. I am not sure this is the best way to write the code though.